frontend/src/app/pagination.ts

```diff
@@ -70,7 +80,7 @@ abstract class PaginatorAbstraction<T, Params extends URLSearchParams> {
    */
   constructor(
     protected api: string,
-    protected http: HttpClient
+    protected http: HttpClient = inject(HttpClient)
```

Comment:
Is this idiomatic and necessary in new Angular?

I thought the use case for `inject` was more to declare `http` as a local member (at the top of the class definition) and assigning it `inject(HttpClient)` there, to avoid over-reliance on constructor parameters declaring members.

Reply:
```suggestion
    protected http: HttpClient = inject(HttpClient)
```

Reply:
@KrisJordan in this case, removing `injected(HttpClient)` causes an error:
```
An argument for 'http' was not provided.
```
I think since this is in a standalone class, Angular does not seem to automatically inject the HTTPClient. So, in order to prevent having to pass the injected `HTTPClient` instance into each paginator object in every service, this small addition directly injects the client into the object.

Now, it might be better to move this out of the constructor and have it as a field - thoughts?

Reply:
(Also, the `;` suggestion causes a syntax error since it is in the constructor, so I did not include!)

Reply:
Ah, I see, right this is outside of dependency injection scope. Makes sense!